### ISSUE TITLE
Doing up the house. We need to clean up the CNC room, and provide it with a functional modern PC.

### ISSUE THREAD

**Issue by MatthewCroughan** [DoESLiverpool/somebody-should#985]
### We need a mission to reclaim the CNC room.

I attempted to use the CNC for two days straight, and  despite acquiring knowledge, and a story to tell the kids about how bad things were, we need to completely rethink the management of this room. I offer to resolve every issue I bring up here as soon as possible, and to the best of my ability.

### The PCs in the CNC room are running core2 duos. This means **slow.** in English.

I recently ran a 12 hour stream of me figuring out how to use the CNC machine and it went down like this:

- They're running core2 duo CPUs that do not have the ability to support Directx11 *and we're trying to run fusion360, industry cad software on them*
- Trying to grab their screens takes up 20% of the cpu, making streaming feasible but not nice using ffmpeg and gdigrab for Windows at 10fps, a simple screengrab turns the room into a sauna
- I need both 32 bit and 64 bit binaries to make this happen, since one machine has a 32 bit cpu in it
- Generating toolpaths takes around 2-3 minutes instead of 2-3 seconds like on a normal modern machine
- One of them hates me

also,
- I had to  use raw planks of wood to create a makeshift desk, to place my laptop and more on during the stream
- I think I have developed RSI from the dual mouse/keyboard KVM setup that exists
Despite this, the stream was quite successful, but not a very comfortable thing to get on with.

I have a 6 thread AMD machine that is more than capable of generating toolpaths in no time at all that I was reliant on for the majority of the work I was saving, before attempting to load it on these wretched machines. I offer it, I will install it in the room and clone the drive of the existing system over via `dd` any day.  

- We can roll back to using the existing setup if anything goes wrong
- I'm capable of doing all of this without any assistance
- I am begging you

### The room is disorganised and chaotic

All the bits, all the collets, all the machines are in ruins. Broken bits are scattered everywhere, and there are dangerous items littered about that need removing, such as some fluorescent lights that were waiting to be stepped on today. Everything needs to be organised properly into containers labelling what exactly they are. 

Leaving the room like this leads to 1 hour tool changes, which are costly on sanity, a limited resource.

I can come up with solutions for organised storage and offer to keep the room clean, organised in tip top shape, since I will be using it quite a bit. Having a friendlier room would encourage people to experiment more with the CNC, it's far simpler than people make it out to be.

I may need help on this one, I will be in need of assistance for planning optimal usage of the vertical space in the room, in mounting shelves should they be discussed to be viable, and also in need of a vesa mount or two in order to kit the room out with a nice slim computing solution.

### What are the issues with the existing proprietary software?

I've heard rumours that the existing proprietary software is tied to the existing hardware. What is the truth in this? I need to know what the reason for this insane setup currently is.

If what I've heard is the case, I can try a lot of things to solve this with virtualization, the software won't have any idea that it's running on another machine if I'm successful with this.

This will soon be irrelevant with the introduction of LinuxCNC onto the machines, @magman2112 and @goatchurchprime have been working for quite some time on building a box capable of driving the unfortunately proprietary German CNC that we have. We need to push forward towards this goal at some point, if it is the case that I am unable to create a virtual environment that tricks the software.

TL;DR the room's a mess, I want to fix it. Let me know when I can get on with it! No more floccinaucinihilipilification, the cnc has feelings too.

**Comment by MatthewCroughan** [comment]
@mdunschen I opt for cloning the disk because it's far less worry about getting everything right, I'll do it onto an SSD of some sort so it's all nice and fast.

**Comment by Sean-anotherone** [comment]
That is good news !  I saw that the Kinetic Software is licensed to DoES liverpool and went looking for a code - when I found none I made the assumption that it was tied to that PC for now.
If there is some better hardware available, particularly if it will run fusion a bit better, that would be awesome.  I think Martin is correct that no-one uses the German CAD suite that came with the machine, we have a postprocessor for fusion and should standardise on that.

Getting ArchiType up and running is already on my list and that will involve most of the stuff that's lying about in the room being removed / used.  

Something that is in a bit of short supply generally is space for storage of "stuff" - a lot of the CNC room stuff is offcuts that are big enough to keep, strange foamy materials for prototyping etc.  There isn't much in the way of places to stow things like this?
Sean

**Comment by MatthewCroughan** [comment]
@Sean-anotherone I'm cleaning the room out entirely now, you'll see something fresh and lovely on maker night. We should make a policy so material stashing and grime doesn't become a problem again. I vow to keep the place clean at least with a quick clean weekly.

**Comment by Sean-anotherone** [comment]
There are a bunch if files on the desktop, including the fusion license info.  The Kinetic software is on a stick in the software folder in the cupbpard.  The only thing to be careful about is transferring the postprocessor to the new machine.
Sean

**Comment by amcewen** [comment]
More tidying and finding proper homes for things sounds great to me!  My one note on that though would be to work out whether the reorganising mentioned in #955 will impact any more permanent storage-building (i.e. ideally don't spend loads of time building storage stuff we'll (maybe) want to re-engineer in a month or two...)

**Comment by MatthewCroughan** [comment]
@amcewen A suggestion me and Jackie discussed was to have a bargain bin style auction of unwanted materials when the build up starts getting too great.

**Comment by jackie1050** [comment]
@mdales has wall-mounted storage for collets here: https://instagram.com/p/BqdWol6HTRK/ which we could adapt to attach to the metal strip.

**Comment by Sean-anotherone** [comment]
I like it.

**Comment by MatthewCroughan** [comment]
@jackie1050 Is it possible we could get some cork to make this happen with? We had loads when we were messing with biomaterial didn't we?

**Comment by mdales** [comment]
I doubt you want to copy it exactly, but the Fusion 360 model I used is here: 

https://a360.co/2PQWjQ4

**Comment by MatthewCroughan** [comment]
@jackie1050 @mdales Thanks lots for this, will get it set up asap.

**Comment by ajlennon** [comment]
Lovely. Nice work. Well done. Can you update the wiki with the current state of affairs. Thanks!

https://github.com/DoESLiverpool/somebody-should/wiki/PC-Systems

**Comment by MatthewCroughan** [comment]
@ajlennon Done this, fixed up a bunch of references. Also added a small comment to re-evaluating the need to run Windows-based machines. Systems such as the Label Printer/Vinyl Cutter pc can be ran easily on a raspberry pi, I suspect the dymo label printing software can be entirely re-implemented in an evening with Python for example, and there may be a driver on Linux for the Roland, as there already is for the label printers. 

https://packages.debian.org/sid/printer-driver-dymo
https://github.com/sk1project/lincutter

**Comment by jackie1050** [comment]
In case this is relevant, I tried using the sublimation printer from my laptop, but the Linux driver didn't seem to have the same quality options as the Windows one.

On 24 Nov 2018 18:10, MatthewCroughan <notifications@github.com> wrote:

@ajlennon<https://github.com/ajlennon> Done this, fixed up a bunch of references. Also added a small comment to re-evaluating the need to run Windows-based machines. Systems such as the Label Printer/Vinyl Cutter pc can be ran easily on a raspberry pi, I suspect the dymo label printing software can be entirely re-implemented in an evening with Python for example, and there may be a driver on Linux for the Roland, as there already is for the label printers.

https://packages.debian.org/sid/printer-driver-dymo
https://github.com/sk1project/lincutter

**Comment by amcewen** [comment]
Ace to see things getting tidied @MatthewCroughan!

The sublimation printer stuff should be spun out as a new issue if it's something we want to do, given that it's nothing to do with the CNC room :-)  (I'm not sure what the user need is for changing things, myself)

**Comment by MatthewCroughan** [comment]
@amcewen Will make an issue for this now. It's not a case of 'if it's not broken don't fix it', it's a case of 'it's not open source, when it should be'. I'm not interested in doing anything if it breaks the user experience though, so I wouldn't worry about that. Everything I'm suggesting is as an adjacent improvement, not a demolishing recreation. Ideally we'd have both options set up for a while before removing the old setup, although since Inkscape is the main application that's used before going ahead with the job (printer driver), I don't see how things could be made that much different, it's just a case of minimalism the setup to work on a raspberry pi that may work forever in a read-only, network-share connected fashion, instead of a machine that is going to die at some point because of its hard drive. Less maintenance needed, etc.

**Comment by Sean-anotherone** [comment]
The new post sounds great - if it can move the tool some +z before drawing the box and then pause for a (non existent) tool change it would be as useful as the "test" button on the lasers.

Sean

**Comment by goatchurchprime** [comment]
PC thing is done and works perfectly.  Great job!  I'm closing this issue, so we can make some new ones that can also be closed.

**Comment by norru** [comment]
hi @goatchurchprime I am attempting to prepare a path to run on the CNC Step using FreeCAD at home and I am struggling to find the Kinetic-NC post-processor (the wili page mentions a "jgt" processor which I couldn't find).

Is this the processor?

https://github.com/goatchurchprime/transition-CAM/blob/master/postprocessor/kineticnc_post.py
